### DISCUSSION TITLE
dns_spoofing.py error 'Client' object has no attribute 'connection'

### DISCUSSION THREAD

**Discussion by nilesingalls** [mitmproxy/mitmproxy#6074] [General]
Hello,
I'm attempting to get this [contrib](https://github.com/mitmproxy/mitmproxy/blob/main/examples/contrib/dns_spoofing.py) script working on mitmproxy 9.0.1 and running into this error
AttributeError: 'Client' object has no attribute 'connection'

I assume this is related to API changes in mitmproxy 7, but I'm not figuring out what I need to change and I can't get earlier versions to build on my instance.  (I had this setup working years ago using mitmproxy 2.0.2)

Please advise, thank you!

Mitmproxy: 9.0.1
Python:    3.10.6
OpenSSL:   OpenSSL 3.0.7 1 Nov 2022
Platform:  Linux-5.19.0-1023-aws-aarch64-with-glibc2.35

**Comment by nilesingalls** [comment]
of course I get it working 5 minutes after asking for help!  Wish I would have asked for help a couple of hours ago.

changed flow.client_conn.connection.get_servername() to flow.client_conn.sni

**Comment by mhils** [comment]
Glad.you figured it out, please send a pull request with the fixes! 😃
